File: logineer/src/main/java/br/com/gabryel/logineer/repository/UserRepository.java
package br.com.gabryel.logineer.repository;

import br.com.gabryel.logineer.entities.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, String> {

    Optional<User> getByEmail(String email);

    boolean existsByEmail(String email);
}

File: logineer/src/main/kotlin/br/com/gabryel/logineer/repository/UserRepository.kt
package br.com.gabryel.logineer.repository

import br.com.gabryel.logineer.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, String> {
    fun getByEmail(email: String): User?

    fun existsByEmail(email: String): Boolean
}